### PR TITLE
Updated build script and moved to more in-house dependencies

### PR DESCRIPTION
**Pull request by SinTh0r4s**

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -32,12 +32,14 @@ jobs:
...
-    - uses: "marvinpinto/action-automatic-releases@latest"
-      with:
-        repo_token: "${{ secrets.GITHUB_TOKEN }}"
-        automatic_release_tag: "latest"
-        prerelease: true
-        title: "Latest Development Build"
-        files: build/libs/*.jar
-            
+
+    - name: Run server for 1 minute
```
Comment (bombcar) [inline .github/workflows/build-and-test.yml:36]: is 1m enough to start runServer, or is that "1m after finished loading"

File: build.gradle [hunk 2]
```diff
@@ -256,7 +324,7 @@ runClient {
 runServer {
     def arguments = []
 
-    if(usesMixins.toBoolean()) {
+    if (usesMixins.toBoolean()) {
```
Comment (bombcar) [inline build.gradle:327]: the space here seems inconsistent

File: build.gradle [hunk 3]
```diff
@@ -366,11 +452,66 @@ task apiJar(type: Jar) {
...
+if (isNewBuildScriptVersionAvailable()) {
+    if (autoUpdateBuildScript.toBoolean()) {
+        updateBuildScript()
+    } else {
+        println("Build script update available! Run 'gradle updateBuildScript'")
+    }
+}
+
+static URL availableBuildScriptUrl() {
+    new URL("https://raw.githubusercontent.com/SinTh0r4s/ExampleMod1.7.10/main/build.gradle")
```
Comment (bombcar) [inline build.gradle:481]: shouldn't the GTNH mods depend on a GTNH fork of this examplemod?

File: gradlew.bat [hunk 4]
```diff
@@ -46,10 +62,9 @@ echo location of your Java installation.
 goto fail
 
 :init
-@rem Get command-line arguments, handling Windowz variants
```
Comment (bombcar) [inline gradlew.bat:49]: my goodness

**Review by bombcar — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (SinTh0r4s) [reply .github/workflows/build-and-test.yml:36]: It should be plenty, if you don't use tons of dependencies

File: build.gradle (hunk 3 above)
Comment (SinTh0r4s) [reply build.gradle:481]: Once it is all done, yes. Right now we might change a lot of things

File: build.gradle (hunk 2 above)
Comment (boubou19) [reply build.gradle:327]: Well, it's in all the scripts i merged, so it'll have to wait the next buildscript update session

**Review by boubou19 — APPROVED** [review]

**Pull request merged**